### ISSUE TITLE
Automatically close SQLite connection without using contextmanager

### ISSUE THREAD

**Issue by JWCook** [requests-cache/aiohttp-client-cache#187]
Follow-up from #173.

Currently, the aiosqlite connection used by `SQLiteBackend` will only close automatically if used with the `CachedSession` contextmanager. Otherwise, the `Thread` object used by aiosqlite remains open and causes the program to hang (unlike the stdlib `sqlite3` module, which works fine with leaving connections open).

### Workarounds
Either use the contextmanager:
```py
async with CachedSession(cache=SQLiteBackend()) as session:
    ...
```

Or explicitly call `close()`:
```py
session = CachedSession(cache=SQLiteBackend())
...
session.close()
```

### Options
1. Reopen and close connection for every request
    * Reliable, but comes with a performance hit. 
2. Schedule close using `SQLiteBackend.__del__()` ([suggested here](https://github.com/requests-cache/aiohttp-client-cache/issues/173#issuecomment-1730206380))
    * As @aaraney mentioned, this isn't guaranteed to complete before the event loop closes.
3. Subclass and modify `aiosqlite.Connection.run()` to accept a signal to immediately close without blocking the event loop or needing to be scheduled
4. Various other means of forcing the thread to close (like raising an exception inside the thread or messing with its internal queue)

**Comment by JWCook** [comment]
@aaraney Give the changes in `main` a try, and let me know if that resolves this issue for you.

**Comment by JWCook** [comment]
These changes are now available in v0.10.
